tests/testthat/test-rvar-print.R

```diff
@@ -1,75 +1,103 @@
-"rvar<1,2>[2,3] mean ± sd:\
-     [,1]         [,2]         [,3]        \
-[1,]  1.5 ± 0.71   5.5 ± 0.71   9.5 ± 0.71 \
-[2,]  3.5 ± 0.71   7.5 ± 0.71  11.5 ± 0.71 ",
-    fixed = TRUE
+  out <- capture.output(print(x, color = FALSE))
+  expect_match(
+    out,
+    regexp = "rvar<2>\\[2,3\\] mean . sd:",
+    all = FALSE
+  )
+  expect_match(
+    out,
+    regexp = "     \\[,1\\]         \\[,2\\]         \\[,3\\]        ",
+    all = FALSE
+  )
+  expect_match(
+    out,
+    regexp = "\\[1,\\]  1.5 . 0.71   5.5 . 0.71   9.5 . 0.71 ",
+    all = FALSE
+  )
+  out <- capture.output(print(x_with_chains, color = FALSE))
+  expect_match(
+    out,
+    regexp = "rvar<1,2>\\[2,3\\] mean . sd:",
+    all = FALSE
+  )
+  expect_match(
+    out,
+    regexp = "     \\[,1\\]         \\[,2\\]         \\[,3\\]        ",
+    all = FALSE
+  )
+  expect_match(
+    out,
+    regexp = "\\[1,\\]  1.5 . 0.71   5.5 . 0.71   9.5 . 0.71 ",
+    all = FALSE
+  )
+  expect_match(
+    out,
+    regexp = "\\[2,\\]  3.5 . 0.71   7.5 . 0.71  11.5 . 0.71",
+    all = FALSE
   )
 })
 
 test_that("basic str.rvar works", {
-  skip_on_os("windows")
   x <- rvar(array(1:24, dim = c(2,3,4)))
   x_with_chains <- rvar(array(1:24, dim = c(2,3,4)), nchains = 2)
 
   expect_output(str(rvar()),
-    " rvar<1>[0] ",
-    fixed = TRUE
-  )
-  expect_output(str(rvar(1:3)),
-    " rvar<3>[1]  2 ± 1",
-    fixed = TRUE
+    " rvar<1>\\[0\\] "
   )
-  expect_output(str(x, vec.len = 5),
-    " rvar<2>[3,4]  1.5 ± 0.71  3.5 ± 0.71  5.5 ± 0.71  7.5 ± 0.71  9.5 ± 0.71 ...",
-    fixed = TRUE
+  out <- capture.output(str(rvar(1:3)))
+  expect_match(
+    out,
+    regexp = " rvar<3>\\[1\\]  2 . 1",
+    all = FALSE
   )
-  expect_output(str(x_with_chains, vec.len = 5),
-    " rvar<1,2>[3,4]  1.5 ± 0.71  3.5 ± 0.71  5.5 ± 0.71  7.5 ± 0.71  9.5 ± 0.71 ...",
-    fixed = TRUE
+  out <- capture.output(str(x, vec.len = 5))
+  expect_match(
+    out,
+    regexp = " rvar<2>\\[3,4\\]  1.5 . 0.71  3.5 . 0.71  5.5 . 0.71  7.5 . 0.71  9.5 . 0.71 ...",
+    all = FALSE
   )
 
   x_with_attrs <- x
   dimnames(x_with_attrs)[1] = list(c("a","b","c"))
   attr(draws_of(x_with_attrs), "foo") = list(1,2)
   attr(x_with_attrs, "bar") = list(1,2)
 
-  expect_output(str(x_with_attrs, vec.len = 5),
-    ' rvar<2>[3,4]  1.5 ± 0.71  3.5 ± 0.71  5.5 ± 0.71  7.5 ± 0.71  9.5 ± 0.71 ...
- - dimnames(*)=List of 2
-  ..$ : chr [1:3] "a" "b" "c"
-  ..$ : NULL
- - attr(draws_of(*), "foo")=List of 2
-  ..$ : num 1
-  ..$ : num 2
- - attr(*, "bar")=List of 2
-  ..$ : num 1
-  ..$ : num 2',
-    fixed = TRUE
+  out <- capture.output(str(x_with_attrs, vec.len = 5))
+  expect_match(
+    out,
+    regexp = " rvar<2>\\[3,4\\]  1.5 . 0.71  3.5 . 0.71  5.5 . 0.71  7.5 . 0.71  9.5 . 0.71 ...",
+    all = FALSE
+  )
+  expect_match(
+    out,
+    regexp = " - dimnames\\(\\*\\)=List of 2",
+    all = FALSE
+  )
+  expect_match(
+    out,
+    regexp = '  \\.\\.\\$ : chr \\[1:3\\] "a" "b" "c"',
+    all = FALSE
```

Comment:
Not matching the entire print here, but I think we can trust that if the first 2 lines are as expected the rest is fine as well. Let me know if you want me to expand.